cfn-resources/network-container/cmd/resource/resource.go

```diff
@@ -3,46 +3,165 @@ package resource
 	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/logger"
+	progressevents "github.com/mongodb/mongodbatlas-cloudformation-resources/util/progressevent"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/validator"
 	"go.mongodb.org/atlas/mongodbatlas"
 )
 
-const (
-	defaultProviderName = "AWS"
-)
+var CreateRequiredFields = []string{constants.ProjectID, constants.RegionName, constants.PubKey, constants.PvtKey, constants.AtlasCIDRBlock}
+var ReadRequiredFields = []string{constants.ProjectID, constants.ID, constants.PubKey, constants.PvtKey}
+var UpdateRequiredFields = []string{constants.ProjectID, constants.ID, constants.PubKey, constants.PvtKey}
+var DeleteRequiredFields = []string{constants.ProjectID, constants.ID, constants.PubKey, constants.PvtKey}
+var ListRequiredFields = []string{constants.ProjectID, constants.PubKey, constants.PvtKey}
+
+// function to validate inputs to all actions
+func validateModel(fields []string, model *Model) *handler.ProgressEvent {
+	return validator.ValidateModel(fields, model)
+}
 
 // Create handles the Create event from the Cloudformation service.
 func Create(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	if errEvent := validateModel(CreateRequiredFields, currentModel); errEvent != nil {
+		return *errEvent, nil
+	}
+
+	client, err := util.CreateMongoDBClient(*currentModel.ApiKeys.PublicKey, *currentModel.ApiKeys.PrivateKey)
+	if err != nil {
+		return handler.ProgressEvent{}, err
+	}
+
+	projectID := currentModel.ProjectId
+
+	containerRequest := &mongodbatlas.Container{}
+
+	if projectID == nil || *projectID == "" {
+		return handler.ProgressEvent{}, fmt.Errorf("error creating network container: `ProjectID` must be set")
+	}
+
+	regionName := currentModel.RegionName
+	if regionName == nil || *regionName == "" {
+		return handler.ProgressEvent{}, fmt.Errorf("`error creating network container: RegionName` must be set")
+	}
+	containerRequest.RegionName = *regionName
+	containerRequest.ProviderName = constants.AWS
+	CIDR := currentModel.AtlasCidrBlock
+	if CIDR == nil || *CIDR == "" {
+		return handler.ProgressEvent{}, fmt.Errorf("error creating network container: `AtlasCidrBlock` must be set")
+	}
+	containerRequest.AtlasCIDRBlock = *CIDR
+	containerResponse, res, err := client.Containers.Create(context.Background(), *projectID, containerRequest)
+	if err != nil {
+		if res.StatusCode == http.StatusConflict {
+			_, _ = logger.Debugf("Container already exists for this group. Try return existing container. err: %v", err)
+			containers, _, err2 := client.Containers.ListAll(context.Background(), *projectID, nil)
+			if err2 != nil {
+				_, _ = logger.Debugf("Error Containers.ListAll err:%v", err)
+				return handler.ProgressEvent{}, fmt.Errorf("error Containers.ListAll err:%v", err)
+			}
+			_, _ = logger.Debugf("containers:%v", containers)
+			first := containers[0]
```

Comment:
[q] what if container is `nil`? I think we may want to check that container has one element